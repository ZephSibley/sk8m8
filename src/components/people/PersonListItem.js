import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import ChatIcon from '@material-ui/icons/Chat';


const PersonListItem = props => {
    // Prop: username; string
    // Prop: avatar; string, url src
    // Prop: bio; string
    // Prop: chat; function, takes username - opens ChatView

    const useStyles = makeStyles(theme => ({
        root: {
          width: '100%',
          maxWidth: 360,
          backgroundColor: theme.palette.background.paper,
        },
        inline: {
          display: 'inline',
        },
    }));

    const classes = useStyles();
    return (
        <React.Fragment>
            <ListItem alignItems="flex-start">
                <ListItemAvatar>
                <Avatar
                    alt={props.username}
                    src={props.avatar}
                    //imgProps={loading='lazy'}
                />
                </ListItemAvatar>
                <ListItemText
                primary={props.username}
                secondary={
                    <Typography
                        component="span"
                        variant="body2"
                        className={classes.inline}
                        color="textPrimary"
                    >
                        {props.bio}
                    </Typography>
                }
                />
                <ChatIcon
                    color='primary'
                    fontSize='large'
                    onClick={props.chat(props.username)}
                />
            </ListItem>
            <Divider variant="inset" component="li" />
        </React.Fragment>
    )
};

export default PersonListItem;