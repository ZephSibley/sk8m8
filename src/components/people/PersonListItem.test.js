import React from 'react';
import renderer from 'react-test-renderer';
import PersonListItem from './PersonListItem';

test('Basic Rendering', () => {
    const component = renderer.create(
        <PersonListItem
            username='Zeph'
            avatar=''
            bio='Hi There!'
            chat={function() {return true}}
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})