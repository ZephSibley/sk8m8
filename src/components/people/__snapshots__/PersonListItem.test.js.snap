// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
    >
      <div
        className="MuiAvatar-root MuiAvatar-colorDefault"
      />
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Zeph
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
      >
        Hi There!
      </p>
    </div>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 9h12v2H6V9zm8 5H6v-2h8v2zm4-6H6V6h12v2z"
      />
    </svg>
  </li>,
  <li
    className="MuiDivider-root MuiDivider-inset"
    role="separator"
  />,
]
`;
